core/src/main/scala/org/apache/spark/storage/TachyonBlockManager.scala

```diff
@@ -95,8 +94,23 @@ private[spark] class TachyonBlockManager() extends ExternalBlockManager with Log
   override def putBytes(blockId: BlockId, bytes: ByteBuffer): Unit = {
     val file = getFile(blockId)
     val os = file.getOutStream(WriteType.TRY_CACHE)
-    os.write(bytes.array())
-    os.close()
+    try {
+      os.write(bytes.array())
+      os.close()
```

Comment:
The logic here is that: if any exception happens in write() or close(),  cancel the block file. And current code is able to cache all exceptions.
The OutStream is realized in Tachyon, not common OutStream in Java.